app/controllers/gems_adoptions_controller.rb

```diff
@@ -0,0 +1,30 @@
+class GemsAdoptionsController < ApplicationController
+  before_action :require_login
+
+  def new
+    @gem_adoption = params[:id]
+    @gem = GemsAdoption.new
```

Comment:
@angela4713 I think probably you inverted the names of the variables here. Probably you mean:

```ruby
@gem = params[:id]
@gems_adoption= GemsAdoption.new
```